test/cmd/sdn.sh

```diff
@@ -2,35 +2,65 @@
 source "$(dirname "${BASH_SOURCE}")/../../hack/lib/init.sh"
 trap os::test::junit::reconcile_output EXIT
 
+# Cleanup cluster resources created by this test
+(
+  set +e
+  oc delete namespace sdn-test-1
+  oc delete namespace sdn-test-2
+  oc delete namespace sdn-test-3
+  oc delete egressnetworkpolicy --all
+  exit 0
+) &>/dev/null
+
 os::test::junit::declare_suite_start "cmd/sdn"
 
+os::test::junit::declare_suite_start "cmd/sdn/clusternetworks"
 os::cmd::expect_success 'oc get clusternetworks'
+# Sanity check that the environment is as expected, or the NetNamespace tests will fail
 os::cmd::expect_success_and_text 'oc get clusternetwork default -o jsonpath="{.pluginName}"' 'redhat/openshift-ovs-multitenant'
-os::cmd::expect_failure_and_text 'oc patch clusternetwork default -p "{\"network\": \"1.0.0.0/8\"}"' 'Invalid value'
-os::cmd::expect_failure_and_text 'oc patch clusternetwork default -p "{\"hostsubnetlength\": 22}"' 'Invalid value'
-os::cmd::expect_failure_and_text 'oc patch clusternetwork default -p "{\"serviceNetwork\": \"1.0.0.0/8\"}"' 'Invalid value'
+os::test::junit::declare_suite_end
 
+os::test::junit::declare_suite_start "cmd/sdn/netnamespaces"
 orig_project="$(oc project -q)"
 
 os::cmd::expect_success 'oc get netnamespaces'
 os::cmd::expect_success_and_text 'oc get netnamespace default -o jsonpath="{.netid}"' '^0$'
-os::cmd::expect_failure 'oc get netnamespace sdn-test'
-os::cmd::expect_success 'oc new-project sdn-test'
-os::cmd::expect_success 'oc get namespace sdn-test'
-os::cmd::try_until_success 'oc get netnamespace sdn-test'
-os::cmd::expect_success_and_not_text 'oc get netnamespace sdn-test -o jsonpath="{.netid}"' '^0$'
-os::cmd::expect_success 'oc delete namespace sdn-test'
-os::cmd::try_until_failure 'oc get netnamespace sdn-test'
+
+os::cmd::expect_success 'oc new-project sdn-test-1'
+os::cmd::expect_success 'oc new-project sdn-test-2'
+os::cmd::expect_success 'oc new-project sdn-test-3'
+os::cmd::try_until_success 'oc get netnamespace sdn-test-1'
+os::cmd::expect_success_and_not_text 'oc get netnamespace sdn-test-1 -o jsonpath="{.netid}"' '^0$'
+orig_vnid1="$(oc get netnamespace sdn-test-1 -o jsonpath='{.netid}')"
+os::cmd::try_until_success 'oc get netnamespace sdn-test-2'
+os::cmd::try_until_success 'oc get netnamespace sdn-test-3'
+
+os::cmd::expect_success 'oadm pod-network join-projects --to=sdn-test-1 sdn-test-2'
+os::cmd::expect_success_and_text 'oc get netnamespace sdn-test-2 -o jsonpath="{.netid}"' "^${orig_vnid1}\$"
+
+os::cmd::expect_success 'oadm pod-network make-projects-global sdn-test-1'
+os::cmd::expect_success_and_text 'oc get netnamespace sdn-test-1 -o jsonpath="{.netid}"' '^0$'
+
+os::cmd::expect_success 'oadm pod-network isolate-projects sdn-test-1'
+os::cmd::expect_success_and_not_text 'oc get netnamespace sdn-test-1 -o jsonpath="{.netid}"' '^0$'
+
+os::cmd::expect_success 'oc delete namespace sdn-test-1'
```

Comment:
We've got these in the cluster cleanup block, I don't think we also need them here.